src/Symfony/Bridge/Twig/Extension/SecurityExtension.php

```
@@ -84,6 +87,19 @@ public function getImpersonatePath(string $identifier): string
         return $this->impersonateUrlGenerator->generateImpersonationPath($identifier);
     }
 
+    public function userIsGranted(UserInterface $user, mixed $role, mixed $object = null, ?string $field = null): bool
+    {
+        if (null === $this->userSecurityChecker) {
+            return false;
```

should this return false or throw an exception saying that this function cannot be used when a UserAuthorizationCheckerInterface is not provided ?

---

This one I'm not sure on. isGranted() returns false currently, which is what I mimicked. I agree that we don't necessarily have to follow the same convention as it's a new function, but potentially it should work the same? I'm open to either approach personally.

---

IMO, I'd rather throw a meaningful exception here as it's more convenient for the developer to understand what's going on.

---

IMO, I'd rather throw a meaningful exception here as it's more convenient for the developer to understand what's going on.

---

It is a departure from other twig functions. @fabpot thoughts on this one?

---

Updated to throw exception